src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs

```diff
@@ -78,44 +90,29 @@ public void ProcessStart_UseShellExecute_OnUnix_ThrowsIfNoProgramInstalled(bool
                 fileToOpen = GetTestFilePath() + ".txt";
                 File.WriteAllText(fileToOpen, $"{nameof(ProcessStart_UseShellExecute_OnUnix_ThrowsIfNoProgramInstalled)}");
             }
-
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX) || GetAllowedProgramsToRun().Any(program => IsProgramInstalled(program)))
             {
-                string[] allowedProgramsToRun = { "xdg-open", "gnome-open", "kfmclient" };
-                foreach (var program in allowedProgramsToRun)
+                using (var px = Process.Start(new ProcessStartInfo { UseShellExecute = true, FileName = fileToOpen }))
                 {
-                    if (IsProgramInstalled(program))
+                    Assert.NotNull(px); // on Windows px is null for some reason
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
                     {
-                        programToOpen = program;
-                        break;
+                        // Assert.Equal(programToOpenWith, px.ProcessName); // on OSX, process name is dotnet for some reason. Refer to #23972
+                        Console.WriteLine($"{nameof(ProcessStart_UseShellExecute_OnUnix_ThrowsIfNoProgramInstalled)}(isFolder: {isFolder}), ProcessName: {px.ProcessName}");
                     }
+                    else
+                    {
+                        Assert.Equal(programToOpen, px.ProcessName);
+                    }
+                    px.Kill();
+                    px.WaitForExit();
+                    Assert.True(px.HasExited);
+                    Assert.Equal(ExitCodeKill, px.ExitCode);
```

Comment:
Not worth changing now, but as a general rule, best to not assert more than is necessary for the test. In this case, by asserting the process was killed, you're presumably testing process.Kill. But if process.Kill had a bug, ideally only a test for process.Kill would fail, not lots of tests. You would assert if your test won't work properly if the kill failed, but this test doesn't care.